patch/libprotobuf-mutator.patch

```diff
@@ -0,0 +1,102 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index c018d45..c5f5868 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -18,6 +18,7 @@ project(LibProtobufMutator CXX)
+ enable_language(C)
+ enable_language(CXX)
+ 
++option(TESTING "Enable test building" ON)
+ option(LIB_PROTO_MUTATOR_DOWNLOAD_PROTOBUF
+        "Automatically download working protobuf" OFF)
+ option(LIB_PROTO_MUTATOR_WITH_ASAN "Enable address sanitizer" OFF)
+@@ -110,15 +111,19 @@ else()
+   include_directories(${CMAKE_CURRENT_BINARY_DIR})
+ endif()
+ 
+-enable_testing()
+-
+-include(googletest)
++if (TESTING)
++  enable_testing()
++  find_package(GTest)
++  if (NOT GTEST_FOUND)
++    include(googletest)
++  endif()
+ 
+-if (NOT LIB_PROTO_MUTATOR_CTEST_JOBS)
+-  ProcessorCount(LIB_PROTO_MUTATOR_CTEST_JOBS)
++  if (NOT LIB_PROTO_MUTATOR_CTEST_JOBS)
++    ProcessorCount(LIB_PROTO_MUTATOR_CTEST_JOBS)
++  endif()
++  add_custom_target(check
```

Comment:
It is required to wrap tests in condition in each subdirectory of `example`, because this part, for example
```CMake
add_executable(expat_example_test
               expat_example_test.cc)
add_dependencies(expat_example_test expat_example)
target_link_libraries(expat_example_test
                      ${GTEST_BOTH_LIBRARIES}
                      ${CMAKE_THREAD_LIBS_INIT})
add_test(test.expat_example_test expat_example_test --gtest_color=yes AUTO)
add_dependencies(check expat_example_test)
```
adds dependencies for check, which is not created when testing is disabled. Check other directories for same code.